pkg/controller/scyllacluster/resource.go

```diff
@@ -552,14 +552,14 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 								},
 							},
 							// Before a Scylla Pod is stopped, execute nodetool drain to
-							// flush the memtable to disk and stop listening for connections.
-							// This is necessary to ensure we don't lose any data and we don't
-							// need to replay the commitlog in the next startup.
+							// flush the memtable to disk, finish existing requests and stop listening for connections.
+							// Sleep is required to give chance to Load Balancers to acknowledge Pod going down with their
+							// probes.
 							Lifecycle: &corev1.Lifecycle{
 								PreStop: &corev1.LifecycleHandler{
 									Exec: &corev1.ExecAction{
 										Command: []string{
-											"/bin/sh", "-c", "PID=$(pgrep -x scylla);supervisorctl stop scylla; while kill -0 $PID; do sleep 1; done;",
+											"/bin/sh", "-c", "nodetool drain; sleep 15",
```

Comment:
I think any sleep here needs to be configurable, given how LB are (not) integrated / the unpredictability of any network. The approach here also depends on what happens when a new connections arrives after the drain. Does it need to be drained again? Does scylla refuse a new connection after drain? Depending on the answer I see 2 algorithms:

a)
1. `sleep <minTerminationPeriod>`
1. `nodetool drain`

b) 
1. `sleep <minTerminationPeriod> &`
1. `nodetool drain &`
1. `wait`

Reply:
synced on call, going with b)